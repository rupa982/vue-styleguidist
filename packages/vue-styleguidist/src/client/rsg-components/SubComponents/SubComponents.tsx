import React, { useState } from 'react'
import clsx from 'clsx'
import PropTypes from 'prop-types'
import * as Rsg from 'react-styleguidist'
import Styled, { JssInjectedProps } from 'rsg-components/Styled'

export interface SubComponentsProps {
	subComponents: { name: string; url: string }[]
}

const styles = ({ space, fontFamily }: Rsg.Theme) => ({
	requires: {
		isolate: false,
		fontFamily: fontFamily.base,
		overflow: 'hidden',
		whiteSpace: 'nowrap',
		textOverflow: 'ellipsis',
		marginBottom: space[3]
	},
	requiresOpen: {
		isolate: false,
		overflow: 'visible',
		whiteSpace: 'normal'
	}
})

export const SubComponents: React.FC<SubComponentsProps & JssInjectedProps> = ({ classes, ...props }) => {
	const [requiresOpen, setOpen] = useState(false)
	return (
		<div className={clsx(classes.requires, requiresOpen && classes.requiresOpen)}>
			<b onClick={() => setOpen(!requiresOpen)}>{requiresOpen ? '-' : '+'} Requires </b>
			{props.subComponents.map((subComponent, i) => (
				<a key={i} href={subComponent.url}>
					{subComponent.name}
				</a>
			))}
		</div>
	)
}

SubComponents.propTypes = {
	subComponents: PropTypes.arrayOf(
		PropTypes.shape({
			name: PropTypes.string.isRequired,
			url: PropTypes.string.isRequired
		}).isRequired
	).isRequired
}

export default Styled<SubComponentsProps & JssInjectedProps>(styles as any)(SubComponents)
