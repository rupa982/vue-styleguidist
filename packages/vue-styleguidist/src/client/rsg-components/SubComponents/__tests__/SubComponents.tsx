import React from 'react'
import { render } from '@testing-library/react'
import SubComponents from '../SubComponents'

describe('JsDoc', () => {
	it('should render Markdown', () => {
		const actual = render(<SubComponents subComponents={[{ name: 'option', url: '#comp/options' }]} />)

		expect(actual.container).toMatchSnapshot()
	})
})
