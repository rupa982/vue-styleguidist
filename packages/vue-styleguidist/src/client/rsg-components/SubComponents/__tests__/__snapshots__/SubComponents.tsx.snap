// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsDoc should render Markdown 1`] = `
<div>
  <div
    class="rsg--root-0 rsg--open-3"
  >
    <b
      class="rsg--title-2"
    >
      
       Requires
      
    </b>
    <a
      class="rsg--link-6 rsg--element-4"
      href="#comp/options"
    >
      option
    </a>
  </div>
</div>
`;
